js/qryptos.js

```diff
@@ -108,20 +108,30 @@ module.exports = class qryptos extends Exchange {
             let taker = this.safeFloat (market, 'taker_fee');
             let active = !market['disabled'];
             let minAmount = undefined;
+            let minPrice = undefined;
             if (base === 'BTC') {
                 minAmount = 0.001;
             } else if (base === 'ETH') {
                 minAmount = 0.01;
-            } else if (base === 'QASH' || base === 'UBTC' || base === 'DASH' || base === 'NEO' || base === 'BCH') {
-                minAmount = 0.0001;
+            } else {
+                minAmount = undefined;
+            }
+            if (quote === 'BTC') {
+                minPrice = 0.00000001;
+            } else if (quote === 'ETH' || quote === 'USD' || quote === 'JPY') {
+                minPrice = 0.00001;
+            } else {
+                minPrice = undefined;
             }
-            let minPrice = 0.00001;
             let limits = {
                 amount: { min: minAmount },
                 price: { min: minPrice },
                 cost: { min: minPrice * minAmount },
             };
-            let precision = { amount: 8, price: 5 };
+            let precision = {
+                amount: (typeof minAmount !== 'undefined') ? -Math.log10 (minAmount) : undefined,
+                price: (typeof minPrice !== 'undefined') ? -Math.log10 (minPrice) : undefined,
```

Comment:
That does not make any sense. if minPrice is 0.00001 then precision['price'] is 5 when it is actually 8. Am I wrong?

Reply:
Yeap, you're right. Partly.

You're right when you sa `limits.price.min` does not directly relate to `precision.price`.

At the same time, you can't probably say it's actually 8... because it varies from market to market.

Unfortunately, qryptos does not provide an API endpoint to fetch market parameters so they need to be hardcoded. Something like that: https://github.com/ccxt/ccxt/blob/e22eba2041a1863bb7e6ae3e76ee2c91897c0578/js/bitmarket.js#L100-L108

The problem is that there are dozens of markets there. And it's a lot of hardcode. At the time I tried to do this it appeared that all of these markets had different parameters so I had to manually guess precisions by fetching trades and check min limits by creating orders with increasing values (price and amounts) until I start to get `InsufficientFunds` instead of `InvalidOrder`.

Basically, I gave up and made a very limited logic (thinking that 'something' is probably better than 'nothing').

It can be improved, of course.

Reply:
Hmmm, this should be easily scrapable (they don't even use a captcha) from their website. I would write a scraping bot tonight but I have an exam tomorrow morning but I will look into this tomorrow anyway. Thanks for letting me know (:

Reply:
> Thanks for letting me know (:

I didn't realize you do scraping, actually... Will mention you next time if I stumble on a similar issue.

> On first glance it seems BTC and ETH markets use a price precision of 8 and QASH markets use a price precision of 4, can you point me to any anomalies?

By fetching trades I see that QASH has 8 digits too. AFAICR anomalies lie in quoinex which is derived from qryptos. They quote in different fiat currencies with different precisions.

Reply:
I would do the scraping in the lib itself, at least, for the market info that would be sufficient for all other methods ) This is something that we already do for certain info in the lib (web-scraping kraken limits, for example).

Reply:
Yeh, I noticed kraken scraping... Still, didn't find a similar qryptos page for that.